### PR TITLE
rnote: init at 0.3.4

### PR DESCRIPTION
**Pull request by yrd**
###### Motivation for this change

Adds a package for [rnote](https://github.com/flxzt/rnote), a desktop drawing application similar to Xournal++.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by dotlambda — COMMENTED** [review]
Please squash the commits.

File: pkgs/applications/graphics/rnote/default.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    substituteInPlace meson_post_install.py --replace "gtk-update-icon-cache" "gtk4-update-icon-cache"
+    popd
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/flxzt/rnote";
+    description = "Simple drawing application to create handwritten notes";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ yrd ];
+    platforms = platforms.gnu ++ platforms.linux;
```
[the remark this answers is not in the document]
Comment (yrd) [reply pkgs/applications/graphics/rnote/default.nix:75]: Got that from some other app and figured it was rust-specific. But I'll remove it if we don't need it.

File: pkgs/applications/graphics/rnote/default.nix [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  src = fetchFromGitHub {
+    owner = "flxzt";
+    repo = "rnote";
+    rev = "v${version}";
+    sha256 = "5g5SQJc5aopYxtHNP5T85TtcazovrveUCnMhJ90p2t4=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}-cargo";
```
Comment (dotlambda) [inline pkgs/applications/graphics/rnote/default.nix:36]: There's not really a need for the suffix cause `-vendor` is added automatically, but it's up to you.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 156699` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>rnote</li>
  </ul>
</details>

File: pkgs/applications/graphics/rnote/default.nix (hunk 2 above)
Comment (yrd) [reply pkgs/applications/graphics/rnote/default.nix]: Oh yeah, now that you mention it...

**Comment by yrd** [comment]
@dotlambda Did the build problem go away?

File: pkgs/applications/graphics/rnote/default.nix [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  nativeBuildInputs = [
+    cargo
+    desktop-file-utils # For update-desktop-database
+    meson
+    ninja
+    pkg-config
+    python3 # For the postinstall script
+    rustPlatform.cargoSetupHook
+    rustc
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/rnote/default.nix:48]:
```suggestion
    rustPlatform.rustc
```
Isn't it supposed to be this?

File: pkgs/applications/graphics/rnote/default.nix [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-vnLesWXdqNzlWNQsUVy03kfmcDNazQ1BbizQDoG1kgM=";
+  };
+
+  nativeBuildInputs = [
+    cargo
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/rnote/default.nix:41]:
```suggestion
    rustPlatform.cargo
```

Comment (yrd) [reply pkgs/applications/graphics/rnote/default.nix:41]: I get `error: attribute 'cargo' missing` here. Other examples in nixpkgs seem to either use `cargo`, `rust.cargo` or `rustPlatform.rust.cargo`. Is there some sort of a default?

**Comment by dotlambda** [comment]
> @dotlambda Did the build problem go away?

There was none, it was a false negative.

**Review by dotlambda — APPROVED** [review]

File: pkgs/applications/graphics/rnote/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/graphics/rnote/default.nix:41]: I think we want to use `rustPlatform.rust.cargo` here to not mix and match rust/cargo versions.

**Pull request merged**